locksmith/src/controllers/v2/magicEdenController.ts

```diff
@@ -0,0 +1,34 @@
+import { z } from 'zod'
+import normalizer from '../../utils/normalizer'
+import networks from '@unlock-protocol/networks'
+import { Request, Response } from 'express'
+import { ethers } from 'ethers'
+
+const PurchaseBody = z.object({
+  address: z.string(),
+})
+
+export const purchase = async (request: Request, response: Response) => {
+  // const _lockAddress = normalizer.ethereumAddress(request.params.lockAddress)
+  const network = Number(request.params.network)
+
+  const { address } = await PurchaseBody.parseAsync(request.body)
+  console.log({ address })
+  const recipient = normalizer.ethereumAddress(address)
+
+  const referrer = normalizer.ethereumAddress(
+    networks[network]?.multisig || recipient
+  )
+
+  // TODO: check version of the lock as it may impact the signature
+  // TODO: simulate tx to identify if it would fail!
+  // We need to return the params for the purchase function
+  const params = {
```

Comment:
Since `_values` is also an input into the `purchase` method, we'll also need it to be returned from the endpoint

Reply:
ha! Good point ok let me add that!